File: backups/research_analytics_backend_20260606_231345/backend/app/schemas/admin_dashboard.py
from pydantic import BaseModel


class FeedbackStats(BaseModel):
    total: int
    average_completion_rate: float


class TemplateUsageStats(BaseModel):
    total_templates: int
    approved_templates: int


class PrescriptionTrendPoint(BaseModel):
    date: str
    generated: int
    published: int


class RuleHitRankItem(BaseModel):
    rule: str
    count: int


class TemplateUsageRankItem(BaseModel):
    template: str
    count: int


class RiskRuleStatus(BaseModel):
    total: int
    active: int


class ActionStatus(BaseModel):
    approved: int
    pending_review: int


class TemplateStatusSummary(BaseModel):
    total: int
    approved: int


class KnowledgeStatus(BaseModel):
    documents: int
    skipped_documents: int = 0
    chunks: int
    indexed_chunks: int


class LLMStatus(BaseModel):
    provider: str
    production_ready: bool
    model: str | None = None
    status: str | None = None


class ComplianceStatus(BaseModel):
    confirmed: int
    draft: int


class ProviderStatus(BaseModel):
    provider: str
    model: str | None = None
    production_ready: bool | None = None
    enabled: bool | None = None
    status: str | None = None


class ReferenceDataStatus(BaseModel):
    risk_rules: RiskRuleStatus
    actions: ActionStatus
    approved_actions_count: int
    pending_actions_count: int
    compliance: ComplianceStatus
    confirmed_compliance_count: int
    draft_compliance_count: int
    templates: TemplateStatusSummary
    knowledge: KnowledgeStatus
    rag_active_documents_count: int
    rag_skipped_documents_count: int
    llm: LLMStatus
    embedding: ProviderStatus
    ocr: ProviderStatus
    embedding_provider: str
    ocr_enabled: bool
    llm_provider: str
    ollama_ready: bool


class AdminDashboardSummary(BaseModel):
    total_users: int
    users_by_role: dict[str, int]
    risk_distribution: dict[str, int]
    prescription_status: dict[str, int]
    review_stats: dict[str, int]
    r2_review_rate: float
    r3_referral_count: int
    feedback_stats: FeedbackStats
    template_usage: TemplateUsageStats
    prescription_trend: list[PrescriptionTrendPoint] = []
    rule_hit_rank: list[RuleHitRankItem] = []
    template_usage_rank: list[TemplateUsageRankItem] = []
    cluster_distribution: dict[str, int]
    reference_data_status: ReferenceDataStatus

File: backups/prod_source_sync_20260607_011051/frontend/src/api/adminDashboard.ts
import { apiClient } from "./client";

export interface AdminDashboardSummary {
  total_users: number;
  users_by_role: Record<string, number>;
  risk_distribution: Record<string, number>;
  prescription_status: Record<string, number>;
  review_stats: Record<string, number>;
  r2_review_rate: number;
  r3_referral_count: number;
  feedback_stats: {
    total: number;
    average_completion_rate: number;
  };
  template_usage: {
    total_templates: number;
    approved_templates: number;
  };
  prescription_trend: Array<{
    date: string;
    generated: number;
    published: number;
  }>;
  rule_hit_rank: Array<{
    rule: string;
    count: number;
  }>;
  template_usage_rank: Array<{
    template: string;
    count: number;
  }>;
  cluster_distribution: Record<string, number>;
  reference_data_status: {
    risk_rules: {
      total: number;
      active: number;
    };
    actions: {
      approved: number;
      pending_review: number;
    };
    approved_actions_count: number;
    pending_actions_count: number;
    compliance: {
      confirmed: number;
      draft: number;
    };
    confirmed_compliance_count: number;
    draft_compliance_count: number;
    templates: {
      total: number;
      approved: number;
    };
    knowledge: {
      documents: number;
      skipped_documents: number;
      chunks: number;
      indexed_chunks: number;
    };
    rag_active_documents_count: number;
    rag_skipped_documents_count: number;
    llm: {
      provider: string;
      model?: string | null;
      production_ready: boolean;
      status?: string | null;
    };
    embedding: {
      provider: string;
      model?: string | null;
      production_ready?: boolean | null;
      status?: string | null;
    };
    ocr: {
      provider: string;
      enabled?: boolean | null;
      status?: string | null;
    };
    embedding_provider: string;
    ocr_enabled: boolean;
    llm_provider: string;
    ollama_ready: boolean;
  };
}

export function getAdminDashboardSummary() {
  return apiClient.get<AdminDashboardSummary>("/admin/dashboard/summary").then((response) => response.data);
}
